### PR TITLE
Transformation stage 1: Remove deprecated collections

### PR DESCRIPTION
**Pull request by pzelasko**
> [!IMPORTANT]  
> The `Update branch` button must only be pressed in very rare occassions.
> An outdated branch is never blocking the merge of a PR.
> Please reach out to the automation team before pressing that button.

# What does this PR do ?

Removed collections: avlm, diffusion, llm, multimodal, multimodal-autoregressive, nlp, speechlm, vision, vlm.

**Collection**: [Note which collection this PR will affect]

# Changelog

- Removed collections: avlm, diffusion, llm, multimodal, multimodal-autoregressive, nlp, speechlm, vision, vlm.

# Usage

- You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"

**Pre checks**:

- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:

- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed.
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information

- Related to # (issue)

### CODE REVIEW

**Comment by pzelasko** [comment]
I think we should ignore CodeQL here, it's flagging stuff that did not change in this PR.

File: nemo/collections/audio/data/data_simulation.py [hunk 1]
```diff
@@ -41,6 +40,13 @@
 except ImportError:
     PRA = False
 
+try:
+    import h5py
+
+    HAS_H5PY = True
+except ImportError:
+    HAS_H5PY = False
```
Comment (blisc) [inline nemo/collections/audio/data/data_simulation.py:48]: Why is this library? Should we just add it to the requirements?

File: nemo/lightning/pytorch/optim/lr_scheduler.py [hunk 2]
```diff
@@ -1,547 +0,0 @@
...
-    PolynomialHoldDecayAnnealing,
-    SquareAnnealing,
-    SquareRootAnnealing,
-    T5InverseSquareRootAnnealing,
-    WarmupAnnealing,
-    WarmupHoldAnnealLinear,
-    WarmupHoldAnnealOneMinusSquareRoot,
-    WarmupHoldPolicy,
-    WarmupPolicy,
-)
```
Comment (blisc) [inline nemo/lightning/pytorch/optim/lr_scheduler.py:31]: Are these outdated now? I thought we still use these

File: tests/lightning/test_one_logger_callback.py [hunk 3]
```diff
@@ -1,686 +0,0 @@
-# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
```
Comment (blisc) [inline tests/lightning/test_one_logger_callback.py:1]: You didn't delete nemo/lightning/one_logger_callback.py so we should not delete it's associated test.

File: nemo/lightning/pytorch/optim/lr_scheduler.py (hunk 2 above)
Comment (pzelasko) [reply nemo/lightning/pytorch/optim/lr_scheduler.py:31]: These are all kept in `nemo/core/optim/lr_scheduler`, we only remove their imports and wrappers in NeMo2 lightning + mcore wrapper (`nemo/lightning`) directory.

File: nemo/collections/audio/data/data_simulation.py (hunk 1 above)
Comment (pzelasko) [reply nemo/collections/audio/data/data_simulation.py:48]: I don't think it's really needed, we could just save this stuff in numpy format.
IMO we should just remove h5py, but I'd rather leave it for a separate PR.

File: tests/lightning/test_one_logger_callback.py (hunk 3 above)
Comment (pzelasko) [reply tests/lightning/test_one_logger_callback.py:1]: Fixed

**Comment by pzelasko** [comment]
Passed all tests so CI and overall structure seems intact. I’m ignoring CodeQL scan as it flags issues not introduced here. LGTM!
